Some improvements

Psychokiller1888 opened the pull request:
Hi there Zurich!

This adds onMessageFrom and proper onMessage while renaming onMessage to onMessageIs. This way, Construct 2 can listen for any messages income and for messages coming in from a specific device. The old onMessage is too restrictive by forcing us to specify deviceId and message

(to airconsole 1.5.0 commit was reverted btw, adding it later, with the missing functions)

Commit:
To AirConsole 1.5.0

Commit:
I don't like editing .md s...

Commit:
Modded onMessage to OnMessageIs
Added onMessage
Added onMessageFrom

Commit:
readme....

Commit:
Added onAnyDeviceLeft

francois-dibulo commented on plugin/files/airconsole/runtime.js:
```diff
@@ -77,7 +77,9 @@ cr.plugins_.AirConsole = function(runtime)
-          self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnMessage, self);
+          self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnMessageIs, self);
+		  self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnMessageFrom, self);
+		  self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnMessage, self);
```
When a message is received I either want to know:
- From which device and which message-data it contains (prev. OnMessage)
- Any device and which message it contains (OnMessageKey)

Like this we trigger OnMessage always.

Also, wouldn't you check anyways what the message-data is if you know that a specific device sent it?
(which in the end is OnMessage). Can you give a use-case?

(Please also check the indentation)

Psychokiller1888 replied:
Well, I want to know if we receive a message, from whoever it is and whatever it contains

http://puu.sh/rM0gJ.png

Original OnMessage restrics  in a way that the message reception only triggers if we know both other conditions, DeviceId and Message. With these two, we can trigger  actions if a specific device sent a message, or if anybody sent anything. Applications? Well, say I build a pictionary game, first to answer wins, my controllers will send strings back to the screen and onMessage will trigger. I can't do that with the old implementation, I need to know the message and the deviceId

francois-dibulo replied:
Ok! Can you rephrase the descriptions to:

OnMessageIs:
  Triggered when a specific message is received from a specific device.

OnMessage:
  Triggered when any message is received from any device

OnMessageFrom:
  Triggered when any message is received from a specific device

Thanks!

francois-dibulo commented on plugin/files/airconsole/runtime.js:
```diff
@@ -86,6 +88,7 @@ cr.plugins_.AirConsole = function(runtime)
       if (!device_data) {
         self.ac_leave_id = device_id;
         self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnDeviceLeft, self);
+		self.runtime.trigger(cr.plugins_.AirConsole.prototype.cnds.OnAnyDeviceLeft, self);
```
Indentation please :)

Psychokiller1888 replied:
Missed that one, sry ;) I've fixed them all for a next PR, containing missing functions and upgrade to 1.5.0

http://puu.sh/rM0An.png
http://puu.sh/rM0FN.png

francois-dibulo replied:
Great! Can you fix those things and then push the new changes with 1.5.0?! Saw you also implemented the highscores! 👍

Psychokiller1888 replied:
Sure, will do. Didn't want to push the highscores, I have no way to test them really. Communication works, triggers do their job, but I don't know about the data received. Need to upload a finished game on your servers, the simulator doesn't seem to accept highscores. Can push as an experimental implementation, marked as deprecated in C2

francois-dibulo commented on plugin/files/airconsole/edittime.js:
```diff
@@ -37,11 +37,11 @@ AddCmpParam("Is", "Compare message data key");
 // AND DEVICE ID IS
 AddAnyTypeParam("Object DeviceID", "The object associated to this device id");
-AddCondition(1, cf_trigger, "On message", "Data", "On message is {0} and from {1}", "Triggered when a message is received from a device.", "OnMessage");
+AddCondition(1, cf_trigger, "On specific message", "Data", "On message is {0} and from {1}", "Triggered when a specific message is received from a device.", "OnMessageIs");
```
Is this backwards compatible?

Psychokiller1888 replied:
Yeah, this is just renaming the function, should be ok. Will try in a few minutes

Psychokiller1888 replied:
Yes it is. Reverted back to master, created anew  project, just added OnMessage, saved, closed, reinjected the updated one, reopened and it's ok

francois-dibulo replied:
👍

francois-dibulo reviewed (commented):
Thanks for the PR! Left some comments for clarification!

Commit:
- Updated to 1.5.0
- Added basic highscores  support, not sure how they works on call back, so WIP for the conditions data
-- Conditions "OnHighScores" and  "OnHighScoreStored"
-- Actions "RequestHighScores" and "StoreHighScores"
- Rephrased some descriptions
- Removed PHPStorm auto edit flags
- Fixed identation where needed (sry for that again)
- Added "NicknameJoin", "Nickname", "ProfilePicture", "ProfilePictureJoin", "DeviceUID" expressions

Psychokiller1888 commented:
Ok, this is my current state

I have one problem though and I suspect C2 to be in fault. When using 1.5.0, we get very often a js error when previewing. Instantiation fails on new AirConsole(). The error usually disappears on a simple F5... The error never happens when exporting the project to html or whatever. So delay problems maybe? Or the dynamic <script> tag generation? I simply don't get it... Works 9 timeson 10 for me, for others it  fails 5 times on 10...

http://puu.sh/rNpJr.png

Don't be afraid, there's more coming, I'm just overwhelmed with work these last days. Let me know if you're nok with my changes

Commit:
OnTooManyPlayers

Commit:
Working onTooManyPlayers

Commit:
Fixes random script loading error. Please refer to 'how DOM is handled'
http://puu.sh/rTEc8.png

Commit:
Packing

Commit:
README
